zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -693,17 +695,32 @@ impl<'a, F: Field> EVMConstraintBuilder<'a, F> {
     }
 
     pub(crate) fn opcode_lookup_rlc(&mut self, opcode: Expression<F>, push_rlc: Expression<F>) {
-        self.opcode_lookup_at_rlc(self.curr.state.program_counter.expr(), opcode, push_rlc);
+        //self.opcode_lookup_at_rlc(self.curr.state.program_counter.expr(), opcode, push_rlc);
+        self.opcode_lookup_at_rlc(
+            self.curr.state.program_counter.expr() + self.program_counter_offset.clone(),
+            opcode,
+            push_rlc,
+        );
+
+        self.program_counter_offset = self.program_counter_offset.clone() * self.condition_expr();
     }
 
     #[cfg(feature = "dual_bytecode")]
     // helper to lookup second bytecode table.
     pub(crate) fn opcode_lookup_rlc2(&mut self, opcode: Expression<F>, push_rlc: Expression<F>) {
+        // self.opcode_lookup_at_rlc2(
+        //     self.curr.state.program_counter.expr() + self.condition_expr(),
+        //     opcode,
+        //     push_rlc,
+        // );
         self.opcode_lookup_at_rlc2(
-            self.curr.state.program_counter.expr() + self.condition_expr(),
+            self.curr.state.program_counter.expr()
+                + self.program_counter_offset.expr() * self.condition_expr(),
             opcode,
             push_rlc,
         );
+
+        self.program_counter_offset = self.program_counter_offset.clone() * self.condition_expr();
```

Comment:
we don't need `program_counter_offset` at all? can delete it totally?

Reply:
removed finally !